tests/compile-fail/invalid_enum_discriminant.rs

```diff
@@ -9,9 +9,10 @@ pub enum Foo {
 fn main() {
     let f = unsafe { std::mem::transmute::<i32, Foo>(42) };
     match f {
-        Foo::A => {}, //~ ERROR invalid enum discriminant value read
+        Foo::A => {},
         Foo::B => {},
         Foo::C => {},
         Foo::D => {},
     }
-}
+} //~ ERROR constant evaluation error [E0080]
```

Comment:
Why here?

Reply:
Not sure why exactly here, but it's delayed because we don't have validation anymore